lib/dry/configurable.rb

```diff
@@ -114,14 +115,30 @@ def _settings
     def _config_for(&block)
       config_klass = ::Class.new { extend ::Dry::Configurable }
       config_klass.instance_eval(&block)
-      config_klass.config
+      create_nested_config(config_klass)
```

Comment:
Could just update `::Dry::Configurable::NestedConfig.new` to:

```ruby
module Dry
  module Configurable
    # @private
    class NestedConfig
      def initialize(&block)
        klass = ::Class.new { extend ::Dry::Configurable }
        klass.instance_eval(&block)
        @klass = klass
      end
      # ...
```

Then it fully encapsulates the creation of nested config